internal/designate/client/client.go

```diff
@@ -104,12 +109,20 @@ func (c designateClient) ForEachZone(ctx context.Context, handler func(zone *zon
 			return true, nil
 		},
 	)
+	duration := time.Since(startTime)
+	metrics.ApiCallLatency.WithLabelValues("ForEachZone").Observe(duration.Seconds())
+	if err != nil {
+		metrics.FailedApiCallsTotal.Inc()
+	}
+	return err
 }
 
 // ForEachRecordSet calls handler for each recordset in the given DNS zone
 func (c designateClient) ForEachRecordSet(ctx context.Context, zoneID string, handler func(recordSet *recordsets.RecordSet) error) error {
+	startTime := time.Now()
+	metrics.TotalApiCalls.Inc()
```

Comment:
Same here, I think the pager abstracts away one ore multiple api calls (retrieving a list initially, but being able to retrieve further pages if the list exceeds page-size), so we cannot really say it was one api call here.